### PR TITLE
BUG: fixup FieldInfoContainer.add_field

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary
A couple fixes for this method. Most importantly fixes some broken leftovers from https://github.com/yt-project/yt/pull/3240

- BUG: fix a broken internal function call
- ENH: clarify FieldInfoContainer.add_field's signature (add missing force_override kwarg)
- ENH: avoid confusing exception chaining in sanitizer method

The bug this fixes isn't present on the backport branch so this gets triaged to 4.1

### CODE REVIEW

**Review by matthewturk — DISMISSED** [review]
Seems fine, although I'm not 100% sure of the usage of the bare asterisk in add field.

**Comment by neutrinoceros** [comment]
> although I'm not 100% sure of the usage of the bare asterisk in add field.

This makes any following argument "keyword only", as in, they cannot be passed positionally. This is in line with the existing behaviour, since the argument is currently only captured as a keyword argument.

see https://peps.python.org/pep-3102/

**Comment by neutrinoceros** [comment]
Just noticed they are other places that could use a very similar patch, switching to draft for a minute

**Review by matthewturk — APPROVED** [review]
lgtm

File: yt/fields/field_info_container.py [hunk 1]
```diff
@@ -267,7 +267,7 @@ def setup_fluid_aliases(self, ftype="gas"):
                 self.alias((ftype, alias), field)
 
     @staticmethod
-    def _sanitize_sampling_type(sampling_type):
+    def _sanitize_sampling_type(sampling_type: str) -> str:
```
Comment (matthewturk) [inline yt/fields/field_info_container.py:270]: Maybe -- not in this PR, or this release cycle -- this could even be an enum!

**Review by matthewturk — APPROVED** [review]
LGTM, no changes needed

**Pull request merged**